### Import Libraries

In [108]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns 

In [109]:
df = pd.read_csv('FinalData.csv')
df.head()

ST_CASE  AGE                                     BODY_TYPNAME  DOA  \
0  202110028   43  Two Wheel Motorcycle (excluding motor scooters)    7   
1  202110063   21  Two Wheel Motorcycle (excluding motor scooters)    7   
2  202110096   30  Two Wheel Motorcycle (excluding motor scooters)    7   
3  202110128   50  Two Wheel Motorcycle (excluding motor scooters)    0   
4  202110132   44  Two Wheel Motorcycle (excluding motor scooters)    0   

          DOANAME  HELM_USE HELM_USENAME   INJ_SEVNAME MAK_MOD  MAKENAME  SEX  \
0   Died at Scene         1          DOT  Fatal Injury   73705  Kawasaki    1   
1   Died at Scene         1          DOT  Fatal Injury   34706       BMW    1   
2   Died at Scene         0      NON_DOT  Fatal Injury   73705  Kawasaki    1   
3  Not Applicable         1          DOT  Fatal Injury   37706     Honda    1   
4  Not Applicable         0      NON_DOT  Fatal Injury   53706    Suzuki    1   

  SEXNAME  STATEnbr State   VPICBODYCLASSNAME   Latitude  Longitude   LAW  
0    Male         1    AL  Motorcycle - Sport  34.112536 -86.867147  FULL  
1    Male         1    AL  Motorcycle - Sport  32.577539 -85.524497  FULL  
2    Male         1    AL  Motorcycle - Sport  33.485328 -86.864297  FULL  
3    Male         1    AL  Motorcycle - Sport  34.841892 -86.709217  FULL  
4    Male         1    AL  Motorcycle - Sport  32.330161 -86.232128  FULL

In [110]:
from sklearn.linear_model import LinearRegression
from sklearn.neural_network import MLPRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split


In [111]:
modelLR = LinearRegression()
modelNN = MLPRegressor()
modelRF = RandomForestRegressor

In [112]:
X = df.iloc[:,5:6]
y=df.iloc [:,10:11]

In [113]:
X

HELM_USE
0             1
1             1
2             0
3             1
4             0
...         ...
14110         0
14111         0
14112         0
14113         0
14114         0

[14115 rows x 1 columns]

In [114]:
y

SEX
0        1
1        1
2        1
3        1
4        1
...    ...
14110    1
14111    1
14112    1
14113    1
14114    1

[14115 rows x 1 columns]

In [115]:
Xtrain, Xtest, ytrain, ytest = train_test_split(X,y, test_size=0.2, random_state=1)

In [116]:
Xtest

HELM_USE
5329         17
6463          5
13633         1
5395          5
13174         1
...         ...
605           0
6420          5
8471          0
12801         1
10149         0

[2823 rows x 1 columns]

In [117]:
ytest

SEX
5329     1
6463     1
13633    1
5395     1
13174    1
...    ...
605      1
6420     1
8471     1
12801    1
10149    1

[2823 rows x 1 columns]

In [118]:
modelLR.fit(Xtrain,ytrain)

LinearRegression()

In [119]:
print(modelLR.score(Xtest,ytest))
print("Model coefficients:", modelLR.coef_)
print("Model intercept:", modelLR.intercept_)


-7.961712918591601e-05
Model coefficients: [[0.00044997]]
Model intercept: [1.09725453]


In [120]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

In [121]:
y_pred = modelLR.predict(Xtest)

In [122]:
mse = mean_squared_error(ytest, y_pred)
mae = mean_absolute_error(ytest, y_pred)

print("Mean Squared Error (MSE):", mse)
print("Mean Absolute Error (MAE):", mae)


Mean Squared Error (MSE): 0.09705554587662463
Mean Absolute Error (MAE): 0.17175395076215833


### Linear Regression Scores (y=Sex)
|Attribute|R-Squared|Model Coefficient|Model Intercept|MSE|MAE|
|---      |---      |---              |---            |---|---|  
|HELM_USE|-7.9617|0.000449|1.0972|0.09705|0.17175|
